File: editor/en/[1704]Determine if String Halves Are Alike.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean halvesAreAlike(String s) {
        int mid = s.length() / 2;
        int countFirst = getCount(s.substring(0, mid));
        int countSecond = getCount(s.substring(mid));
        return countFirst == countSecond;
    }

    private int getCount(String s) {
        int count = 0;

        for (char c : s.toCharArray()) {
            if ("aeiouAEIOU".indexOf(c) != -1) {
                count++;
            }
        }
        return count;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1704]Determine if String Halves Are Alike.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function (s) {
    const mid = s.length / 2;
    const vowels = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
    const getCount = (str, vowels) => {

        let count = 0;
        for (const c of str) {
            if (vowels.has(c)) {
                count++;
            }
        }

        return count;
    }

    const first = getCount(s.substring(0, mid), vowels);
    const second = getCount(s.substring(mid), vowels);
    return first === second;
};
//leetcode submit region end(Prohibit modification and deletion)
